unlock-app/src/utils/url.ts

```diff
@@ -31,3 +31,13 @@ export const getURL = (url?: string) => {
     return
   }
 }
+
+export const getSlugParamsFromUrl = (url: string) => {
+  const [hash, query] = url?.split('#')[1]?.split('?') ?? []
+  const params = Object.fromEntries(new URLSearchParams(query))
```

Comment:
Why not use URL object then get the it from there. Should validate it is a valid URL too.